### PR TITLE
Add SingleArticleSelection

### PR DESCRIPTION
**Pull request by luca-rath**
| Q | A
| --- | ---
| Bug fix? | no
| New feature? | yes
| BC breaks? | no
| Deprecations? | no
| Fixed tickets | fixes #issuenum
| Related issues/PRs | #issuenum
| License | MIT

#### What's in this PR?

This PR adds a `single_article_selection`

#### Why?

It doesn't exist yet

#### Example Usage

~~~xml
<property name="my_article" type="single_article_selection">
    <meta>
        <title lang="en">My Article</title>
    </meta>
</property>
~~~

#### To Do

- [x] Add documentation page

### CODE REVIEW

File: Controller/ArticleController.php [hunk 1]
```diff
@@ -159,8 +159,18 @@ public function cgetAction(Request $request)
             $search->addQuery($boolQuery);
         }
 
-        if (null !== ($type = $request->get('type'))) {
```
Comment (wachterjohannes) [inline Controller/ArticleController.php]: document the BC break in the UPGRADE.md file

File: Content/SingleArticleSelectionContentType.php [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+    /**
+     * @param Manager $searchManager
+     * @param ReferenceStoreInterface $referenceStore
+     * @param string $articleDocumentClass
+     */
+    public function __construct(
+        Manager $searchManager,
+        ReferenceStoreInterface $referenceStore,
+        $articleDocumentClass
```
Comment (wachterjohannes) [inline Content/SingleArticleSelectionContentType.php]: string $articleDocumentClass

File: Content/SingleArticleSelectionContentType.php [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+     */
+    private $referenceStore;
+
+    /**
+     * @var string
+     */
+    private $articleDocumentClass;
+
+    /**
+     * @param Manager $searchManager
```
Comment (wachterjohannes) [inline Content/SingleArticleSelectionContentType.php]: remove phpdoc

File: Content/SingleArticleSelectionContentType.php [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+            return null;
+        }
+
+        $repository = $this->searchManager->getRepository($this->articleDocumentClass);
+        $locale = $property->getStructure()->getLanguageCode();
+
+        return $repository->find($this->getViewDocumentId($uuid, $locale)) ?? null;
+    }
+
+    /**
```
Comment (wachterjohannes) [inline Content/SingleArticleSelectionContentType.php]: remove phpdoc

File: Content/SingleArticleSelectionContentType.php [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+        $articleDocumentClass
+    ) {
+        parent::__construct('Article');
+
+        $this->searchManager = $searchManager;
+        $this->referenceStore = $referenceStore;
+        $this->articleDocumentClass = $articleDocumentClass;
+    }
+
+    /**
```
Comment (wachterjohannes) [inline Content/SingleArticleSelectionContentType.php]: remove phpdoc

**Comment by wachterjohannes** [comment]
@luca-rath the pipeline is failing

**Comment by luca-rath** [comment]
@wachterjohannes 

> @luca-rath the pipeline is failing

Because i already set min requirement for `sulu/sulu` to `2.0.3` ... the pipeline should pass as soon as the sulu release has been made.

**Pull request merged**